src/main/webapp/SearchResults.js

```diff
@@ -29,162 +59,67 @@ function loading() {
-          ' architecture, Afro-Brazilian culture and a tropical coastline.',
-      id: 203,
-    },
-  ];
-  displayResults(results);
-}
-
-/**
- * Loads South Korea info to page.
- */
-function southKoreaResults() {
-  const headingContainer = document.getElementById('Heading');
-  headingContainer.innerText = 'SOUTH KOREA';
-
-  const results = [
-    {
-      country: 'South Korea',
-      name: 'Seoul',
-      description: 'Seoul, the capital of South Korea,' +
-          ' is a huge metropolis where modern skyscrapers, high-tech subways' +
-          ' and pop culture meet Buddhist temples, palaces and street markets.',
-      id: 201,
-    },
-    {
-      country: 'South Korea',
-      name: 'Busan',
-      description: 'Busan, a large port city in South' +
-          ' Korea, is known for its beaches, mountains and temples.',
-      id: 202,
-    },
-    {
-      country: 'South Korea',
-      name: 'Jeju-do',
-      description: 'Jeju province encompasses the ' +
-          'South Korean island of Jeju in the Korea Strait. Its known for' +
-          'its beach resorts and volcanic landscape of craters and cavelike' +
-          ' lava tubes.',
-      id: 203,
-    },
-  ];
-  displayResults(results);
-}
-
 /**
  * @param {results} results
  */
-function displayResults(results) {
-  // Create a flex-container div if necessary or grab last flex-container
+function displayCityResults(results, countries) {
   const flexElement = document.getElementById('flex-container');
 
   for (let result = 0, l = results.length; result < l; ++result) {
-    // Create a container div
+    if (results[result] !== undefined) {
     const divElement = document.createElement('div');
 
-    // Place image in div container
     const imgUrl = 'images.jpg';
     const imgElement = document.createElement('img');
     imgElement.src = imgUrl;
 
     divElement.appendChild(imgElement);
 
-    // Create a container div
     const divInfoElement = document.createElement('div');
     divInfoElement.className = 'container';
 
-    // populate city name
     const cityElement = document.createElement('h1');
     cityElement.innerText = results[result]['name'];
 
-    // populate country name
+    var countryName ="";
+    for (let country in countries) {
+        if(countries[country]['id'] === results[result]['countryId']){
+          var countryName = countries[country]['name'];
+          break;
+        }
+    }
+
     const countryElement = document.createElement('h2');
-    countryElement.innerText = results[result]['country'];
+    countryElement.innerText = countryName;
 
     const cityDescElement = document.createElement('p');
     cityDescElement.innerText = results[result]['description'];
 
-    // populate city information
     const exploreButtonElement = document.createElement('button');
     exploreButtonElement.className = 'button';
     exploreButtonElement.innerText = 'EXPLORE';
 
```

Comment:
Make the button take the user to the corresponding city/country page. You'll need to set the `onclick` property to a function that sets `location.href` to the corresponding link calculated from the result. I'm guessing based on #21 that @parth-vora intends to have these pages be `city.html` and `country.html` with query parameter `id` so I think you need to add:
```javascript
    let link = "";
    if (result.type == "Country") {
      link = `/country.html?id=` + result.id;
    } else {
      link = '/city.html?id=' + result.id;
    }
    exploreButtonElement.onclick = function(){location.href = link};
```